flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestEnvironment.java

```diff
@@ -201,4 +202,15 @@ protected NetworkFailuresProxy createProxy(String remoteHost, int remotePort) {
 		networkFailuresProxies.add(proxy);
 		return proxy;
 	}
+
+	protected void maybePrintDanglingThreadStacktrace(String threadNameKeyword) {
```

Comment:
I'm wondering about the benefit. What do we gain by knowing where the `KafkaAdminClient` is blocking?

Reply:
This helps to identify where exactly internal thread of `KafkaAdminClient` blocks. So we can address the root cause of the long closure, though it may be in Kafka instead of Flink.

Reply:
Seems unlikely to ever be something in Flink though? I wouldn't want people start chasing Kafka bugs.

Reply:
I think our goal is to make our tests stable, regardless of whether the bug is in Kafka or not. Adding this log is not only for debugging Kafka.

A dangling thread may not necessarily indicate a bug in Kafka. We don't know yet. For example, the blocking could be caused by our test environment setting, e.g. forget to unblock the broker traffic in some case.

Even if it is a bug in Kafka, it is better to know what exactly the bug is so we know the consequence of it. For example, in this case a topic may not be deleted, which may  have impact in other tests.